BHoM/BHoMAnalytics_Toolkit
# Collect usage data from files in the Log folder and send them to a database

### NOTE: Depends on 
https://github.com/BHoM/BHoM_UI/pull/295

   
### Issues addressed by this PR


Closes #1

There's two sections to the new toolkit:
- The engine is in charge of collecting the usage data from the log files (and delete them afterwards if requested). as any other engine method, it is available in the UI. Be aware that it will never read the data from the current log, only from previous log files.
- The adapter isn't really a typical adapter at the moment as it doesn't implement any push or pull method. It is purely there to provide the capability to send the usage data to the database when the UI starts. In the long run, we might start adding pulling functionality to get analytical information back.

### Test files

#### To test that the Engine is correctly collecting data:

![image](https://user-images.githubusercontent.com/16853390/87757171-c5c77e80-c83c-11ea-9af9-aaf0ae33497c.png)

The default value for deleting the processed files false so don't worry :-)

Note that you might have to recompile BHoM_UI and Grasshopper since it looks like the code that creates the log files is not in the installer yet.

if you want to check your log files, you can find them in `C:\ProgramData\BHoM\Logs`

#### To test sending the data to a database:

![image](https://user-images.githubusercontent.com/16853390/87846641-e3135000-c903-11ea-8b52-0642c59f9781.png)

Once you have done that, the toolkit should start sending data to the database at every start of the UI. Be aware that it will delete the processed log files.
Obviously, you might want to switch to the London server or you localhost instead 😄
